platform/darwin/test/MGLStyleValueTests.swift

```diff
@@ -19,41 +19,62 @@ extension MGLStyleValueTests {
+                XCTAssertEqual(actualFunction.defaultValue, expectedFunction.defaultValue)
+            } else if let actualFunction = actualFunction as? MGLCompositeStyleFunction<MGLColor>,
+                let expectedFunction = expectedFunction as? MGLCompositeStyleFunction<MGLColor> {
                 XCTAssertEqual(actualFunction.defaultValue, expectedFunction.defaultValue)
             }
-            guard let actualStops = actualFunction.stops else {
-                XCTAssertNil(expectedFunction.stops)
-                return
-            }
-            let expectedStops = expectedFunction.stops!
-            let actualKeys = actualStops.keys.sorted(by: { String(describing: $0) < String(describing: $1) })
-            let expectedKeys = expectedStops.keys.sorted(by: { String(describing: $0) < String(describing: $1) })
-            XCTAssertEqual(actualKeys, expectedKeys)
             
-            for (ak, ek) in zip(actualKeys, expectedKeys) {
-                XCTAssertEqual(String(describing: ak), String(describing: ek))
-                if let actualValue = actualStops[ak] as? MGLStyleConstantValue<MGLColor> {
-                    guard let expectedValue = expectedStops[ek] as? MGLStyleConstantValue<MGLColor> else {
-                        XCTFail("Unexpected stop \(ak), \(actualValue).  Expected one of \(expectedStops).")
-                        return
-                    }
+            func assertStopEqual (_ actualValue: Any?, _ expectedValue: Any?) {
+                guard type(of: actualValue) == type(of: expectedValue) else {
+                    XCTFail("Expected stop value of type \(type(of: expectedValue)), but found \(type(of: actualValue)) instead.")
+                    return
+                }
+                if let actualValue = actualValue as? MGLStyleConstantValue<MGLColor>,
+                    let expectedValue = expectedValue as? MGLStyleConstantValue<MGLColor> {
                     assertColorsEqualWithAccuracy(actualValue.rawValue, expectedValue.rawValue)
-                } else if let actualValue = actualStops[ak] as? MGLStyleConstantValue<AnyObject> {
-                    let expectedValue = expectedStops[ek] as? MGLStyleConstantValue<AnyObject>
+                } else if let actualValue = actualValue as? MGLStyleConstantValue<AnyObject>,
+                    let expectedValue = expectedValue as? MGLStyleConstantValue<AnyObject> {
                     XCTAssertEqual(actualValue, expectedValue)
                 } else {
-                    XCTFail("Unsupported stop value type \(type(of: actualStops[ak])).")
+                    XCTFail("Unsupported stop value type \(type(of: actualValue)).")
                 }
             }
+            
+            XCTAssertEqual(actualStops.count, expectedStops.count)
+            if let actualStops = actualStops as? [String:Any], let expectedStops = expectedStops as? [String: Any] {
+                actualStops.forEach({ assertStopEqual($1, expectedStops[$0]) })
```

Comment:
Nit: when a function call ends with a closure, you can move it to after the argument list (after the parentheses), and you can also remove the parentheses if there are no other arguments. This is called [trailing closure syntax](https://developer.apple.com/library/prerelease/content/documentation/Swift/Conceptual/Swift_Programming_Language/Closures.html#//apple_ref/doc/uid/TP40014097-CH11-ID102):

```swift
actualStops.forEach { assertStopEqual($1, expectedStops[$0]) }
```

One way to think about trailing closure syntax is that it effectively allows you to define your own custom control statement as a function that takes a closure.

In this case, you aren’t really making use of any functional programming, so it’s clearer to read (and easier to debug) a standard `for` loop:

```swift
for (actual, expected) in actualStops {
    assertStopEqual(actual, expectedStops[expected])
}
```